packages/teams-js/src/internal/utils.ts

```diff
@@ -444,3 +444,26 @@ export function getBase64StringFromBlob(blob: Blob): Promise<string> {
     reader.readAsDataURL(blob);
   });
 }
+
+/**
+ *  Returns an SSR safe reference to the window object
+ * @returns Window object reference
+ */
+
+export function ssrSafeWindow(): Window {
+  if (!inServerSideRenderingEnvironment()) {
+    return window;
+  } else {
+    // This should NEVER actually be written.
+    // If you EVER see this error in ANY log file, something has gone horribly wrong and a bug needs to be filed.
+    throw new Error('window object undefined at SSR check');
+  }
+}
+
+/**
+ * Checks if running in a Server Side Environment
+ * @returns True if running in a Server Side Environment
+ */
+export function inServerSideRenderingEnvironment(): boolean {
```

Comment:
>inServerSideRenderingEnvironment [](http://example.com/codeflow?start=16&length=32)

Is there ever a reason for any code to call this anymore? Or does using `ssrSafeWindow` cover all cases? I noticed that there were still references to `inServerSideRenderingEnvironment` in some parts of the code; wondering if they should just be removed and leverage only `ssrSafeWindow`

Reply:
I figured I would leave `inServerSideRenderingEnvironment` in case someone wanted to use it for anything else aside from a precursor check to referencing `window`. I noticed a couple places that the call could be removed since its just being used as the precursor check, which `ssrSafeWindow` handles, but I found a couple other places its being used differently, so I felt it was worth keeping.